blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/FavoriteContainerImpl.java

```diff
@@ -58,4 +62,15 @@ public BlueFavorite get(String name) {
     public Link getLink() {
         return self;
     }
+
+    /**
+     * Delete all of the user's favorites.
+     */
+    @WebMethod(name = "") @DELETE
+    public void doDelete(StaplerResponse resp) throws Favorites.FavoriteException {
+        for (final Item favorite: Favorites.getFavorites(user.user)) {
```

Comment:
I think you can replace this loop with a Java 8 oneliner http://www.deadcoderising.com/java-8-no-more-loops/

Reply:
`Favorites.getFavorites` returns an `Iterator`, so I think I'd need to do some `Spliterator` junk to get a stream and then operator on the items. I am not sure that's actually more readable? But if you know a better way I am all ears.

Reply:
Looks like I could use this:

```
StreamSupport.stream(Favorites.getFavorites(user.user).spliterator(), false)
            .forEach(favorite -> Favorites.removeFavorite(user.user, favorite));
```

Although unfortunately `Favorites.removeFavorite` throws a checked exception, so I'd have to add a try/catch within the lambda... or change to it to a runtime exception.

Reply:
bloody iterators and streams 👎  not sure why they made this harder and worse to read than guava. 

DW about it

Reply:
We could probably write our own little util class to make it a touch more readable

`StreamHelper.stream(Favorites.getFavorites(user.user))`

where

`public stream(Iterable iterable) { return StreamSupport.stream(iterable.spliterator(), false); }` 

what do you think about the favorites exception being checked? I am not really sure those errors are something we should try to force the user to recover from. I can look at the API more in subsequent PR to see if can be cleaned up?

Reply:
I think its ok to get rid of the checked exception if you want to do that.

Reply:
I have some more work to do in a follow-up PR, so I'll implement it then.